test/integration/camera_sensor.cc

```diff
@@ -2040,4 +2040,230 @@ TEST_F(CameraSensor, SetCompositorNames)
+          std::placeholders::_4, std::placeholders::_5));
+  event::ConnectionPtr c2 =
+    camSensorPincushionNew->Camera()->ConnectNewImageFrame(
+        std::bind(&::OnNewCameraFrame, &imageCount2, img2,
+          std::placeholders::_1, std::placeholders::_2, std::placeholders::_3,
+          std::placeholders::_4, std::placeholders::_5));
+  event::ConnectionPtr c3 =
+    camSensorBarrelLegacy->Camera()->ConnectNewImageFrame(
+        std::bind(&::OnNewCameraFrame, &imageCount3, img3,
+          std::placeholders::_1, std::placeholders::_2, std::placeholders::_3,
+          std::placeholders::_4, std::placeholders::_5));
+  event::ConnectionPtr c4 =
+    camSensorBarrelNew->Camera()->ConnectNewImageFrame(
+        std::bind(&::OnNewCameraFrame, &imageCount4, img4,
+          std::placeholders::_1, std::placeholders::_2, std::placeholders::_3,
+          std::placeholders::_4, std::placeholders::_5));
+
+  // Get some images
+  while (
+    imageCount < 10 || imageCount2 < 10 ||
+    imageCount3 < 10 || imageCount4 < 10)
+  {
+    common::Time::MSleep(10);
+  }
+
+  // Compare colors. Barrel distorted image should have more darker pixels than
+  // the original as the ground plane has been warped to occupy more of the
+  // image. The same should be true for pincushion distortion, because the
+  // ground plane is still distorted to be larger - just different parts
+  // of the image are distorted.
+  unsigned int colorSum1 = 0;
+  unsigned int colorSum2 = 0;
+  unsigned int colorSum3 = 0;
+  unsigned int colorSum4 = 0;
+  unsigned int dumpsterRow = height / 2;
+  for (unsigned int x = 0; x < width*3; x+=3)
+  {
+    unsigned int r1 = img[(dumpsterRow*width*3) + x];
+    unsigned int g1 = img[(dumpsterRow*width*3) + x + 1];
+    unsigned int b1 = img[(dumpsterRow*width*3) + x + 2];
+    colorSum1 += r1 + g1 + b1;
+    unsigned int r2 = img2[(dumpsterRow*width*3) + x];
+    unsigned int g2 = img2[(dumpsterRow*width*3) + x + 1];
+    unsigned int b2 = img2[(dumpsterRow*width*3) + x + 2];
+    colorSum2 += r2 + g2 + b2;
+    unsigned int r3 = img3[(dumpsterRow*width*3) + x];
+    unsigned int g3 = img3[(dumpsterRow*width*3) + x + 1];
+    unsigned int b3 = img3[(dumpsterRow*width*3) + x + 2];
+    colorSum3 += r3 + g3 + b3;
+    unsigned int r4 = img4[(dumpsterRow*width*3) + x];
+    unsigned int g4 = img4[(dumpsterRow*width*3) + x + 1];
+    unsigned int b4 = img4[(dumpsterRow*width*3) + x + 2];
+    colorSum4 += r4 + g4 + b4;
+  }
+
+  // Check that the legacy mode distorts the pincuchin images less
+  EXPECT_GT(colorSum1, colorSum2+800);
+  // Check that there is a good difference in the barrel images
+  // this difference is largely caused by the edges of the new
+  // distortion model which appear gray when they have no value
+  // and gray (192) has a much lower value than white (765)
+  EXPECT_GT(colorSum3, colorSum4+20000);
+
+  // Check that no cropping occurs - specifically the corners
```

Comment:
> "Check that no cropping occurs"

I think this isn't quite right. The legacy barrel image is actually cropped, which is why its corner pixel is white. The pincushion images isn't cropped. Maybe "Check that the corners contain no unrendered pixels"?